src/dstore/pmix_esh.c

```diff
@@ -2245,30 +2242,21 @@ static int _store_data_for_rank(ns_track_elem_t *ns_info, pmix_rank_t rank, pmix
      * so unpack these buffers, and then unpack kvals from each modex buffer,
      * storing them in the shared memory dstore.
      */
-    cnt = 1;
     free_offset = get_free_offset(datadesc);
-    while (PMIX_SUCCESS == (rc = pmix_bfrop.unpack(buf, &bptr, &cnt, PMIX_BUFFER))) {
-        cnt = 1;
-        kp = PMIX_NEW(pmix_kval_t);
-        while (PMIX_SUCCESS == (rc = pmix_bfrop.unpack(bptr, kp, &cnt, PMIX_KVAL))) {
-            pmix_output_verbose(2, pmix_globals.debug_output,
-                                "pmix: unpacked key %s", kp->key);
-            if (PMIX_SUCCESS != (rc = pmix_sm_store(ns_info, rank, kp, &rinfo, data_exist))) {
-                PMIX_ERROR_LOG(rc);
-                return rc;
-            }
-            PMIX_RELEASE(kp); // maintain acctg - hash_store does a retain
-            cnt = 1;
-            kp = PMIX_NEW(pmix_kval_t);
-        }
-        cnt = 1;
-        PMIX_RELEASE(kp);
-        PMIX_RELEASE(bptr);  // free's the data region
-        if (PMIX_ERR_UNPACK_READ_PAST_END_OF_BUFFER != rc) {
+    bptr = buf;
```

Comment:
@karasevb as a minor cleanup - it seems that we don't need `bptr` anymore and `buf` itself should be fine.